.github/workflows/test.yaml

```diff
@@ -128,28 +130,37 @@ jobs:
           pip freeze
           pip check
 
-      - name: Check the Notebook Server extension is installed
-        if: matrix.jupyterlab-version != '4'
+      - name: Check server extension for jupyter_server
+        run: |
+          jupyter server extension list
+          jupyter server extension list 2>&1 | grep -iE "jupyter_server_proxy.*OK" -
+
+      - name: Check server extension for notebook v6
+        if: matrix.notebook-version == '6'
         run: |
           jupyter serverextension list
           jupyter serverextension list 2>&1 | grep -iE "jupyter_server_proxy.*OK" -
 
-      - name: Check the Jupyter Server extension is installed
+      - name: Check frontend extension for notebook v6
+        if: matrix.notebook-version == '6'
         run: |
-          pip install jupyter-server
-          jupyter server extension list
-          jupyter server extension list 2>&1 | grep -iE "jupyter_server_proxy.*OK" -
+          jupyter nbextension list
+          PYTHONUNBUFFERED=1 jupyter nbextension list 2>&1 | grep -A1 -iE '.*jupyter_server_proxy.*enabled' | grep -B1 -iE "Validating.*OK"
+
+      - name: Check frontend extension for notebook v7+
+        if: matrix.notebook-version != '6'
+        run: |
+          jupyter notebook extension list
+          jupyter notebook extension list 2>&1 | grep -iE 'jupyter_server_proxy.*OK.*'
 
-      - name: Check the lab extension
+      - name: Check frontend extension for jupyterlab
         run: |
-          jupyter labextension list
-          jupyter labextension list 2>&1 | grep -iE '@jupyterhub/jupyter-server-proxy.*OK.*'
+          jupyter lab extension list
+          jupyter lab extension list 2>&1 | grep -iE 'jupyter_server_proxy.*OK.*'
 
       - name: Run tests
         run: |
-          mkdir build
-          cd build
-          pytest -vv ../tests -c ../pyproject.toml
+          pytest
```

Comment:
having a look at the coverage, this change results in finding the "at rest" `jupyter_server_proxy` folder, _not_ the as-installed one, during tests. while not a _huge_ deal, it screws up a bunch of the metrics.

while it's annoying to get coverage out to the github issue itself without tokens, we _can_ make a markdown preview and write it out to the ci summary.

some ways around this 
- `rm -rf jupyter_server_proxy` before invoking `pytest`
- adopt the `src/jupyter_server_proxy` layout

Reply:
example: https://github.com/jupyterhub/jupyter-server-proxy/actions/runs/6263255269/job/17007322266?pr=425#step:13:111

Reply:
Yikes ooops I pushed to this branch by mistake =/

I meant to try if this worked still in a private branch in my fork because I didn't understand it why it was done.

What do you think about not doing this tested change and just adding a comment?

Reply:
I've reverted that commit and added a comment as I understand it